cmd/ci-images-mirror/main.go

```diff
@@ -322,6 +338,94 @@ func main() {
 	logrus.Info("Process ended gracefully")
 }
 
+func validateConfig(config *quayiociimagesdistributor.CIImagesMirrorConfig, registryConfig string, ciOperatorConfigPath string, configFilePathInReleaseRepo string) error {
+	diff, err := getNewSupplementalCIImageTargets(configFilePathInReleaseRepo, config)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to get config diff")
+	}
+	errs := []error{}
+	promotedTags := getPromotedTags(ciOperatorConfigPath)
+	for target := range diff {
+		source := config.SupplementalCIImages[target]
+		if source.Image != "" {
+			if strings.HasPrefix(source.Image, "docker.io/") {
+				errs = append(errs, fmt.Errorf("source image %s is from docker.io", source.Image))
+			}
+			if !isAccessible(source.Image, registryConfig) {
+				errs = append(errs, fmt.Errorf("source image %s is not accessible", source.Image))
+			}
+		}
+		if isPromoted(target, promotedTags) {
```

Comment:
Would `promotedTags.has(target)` do the job?

Reply:
Professional deformation 🤦🏼